### PR TITLE
Fix failure cases

### PR DESCRIPTION
**Pull request by MrMarCode**

### CODE REVIEW

File: tests/indent-alignment.test.js [hunk 1]
```diff
@@ -1034,6 +1034,31 @@ describe('Indent Alignment', () => {
       ]);
    });
 
+   it('reports no errors for multi-lined html tags', async () => {
```
Comment (onebytegone) [inline tests/indent-alignment.test.js]:
```suggestion
   it('reports no errors for multi-line html tags', async () => {
```

File: tests/indent-alignment.test.js [hunk 2]
```diff
@@ -1034,6 +1034,31 @@ describe('Indent Alignment', () => {
...
+      await testValidExample([
+         '<a class="this is a long attribute which needs to be wrapped" ',
+         'href="#">This query</a>',
+         'testing',
+      ]);
```
Comment (onebytegone) [inline tests/indent-alignment.test.js]: Can we add tests for other variations of multi-line tags? For example:

```html
<div class="this is a long attribute which needs to be wrapped">
   Lorem ipsum dolor sit amet
</div>
```

File: tests/indent-alignment.test.js [hunk 3]
```diff
@@ -1034,6 +1034,31 @@ describe('Indent Alignment', () => {
       ]);
    });
 
+   it('reports no errors for multi-lined html tags', async () => {
+      await testValidExample([
+         '<a class="this is a long attribute which needs to be wrapped" ',
```
Comment (onebytegone) [inline tests/indent-alignment.test.js]:
```suggestion
         '<a class="this is a long attribute which needs to be wrapped"',
```

File: tests/indent-alignment.test.js [hunk 4]
```diff
@@ -1034,6 +1034,31 @@ describe('Indent Alignment', () => {
...
+         '>',
+         '>             1. test',
+      ]);
+      await testValidExample([
+         'test',
+         '',
+         '                      * test',
+         '',
+         '             1. test',
+      ]);
```
Comment (onebytegone) [inline tests/indent-alignment.test.js]: It's a little weird to call these tests "valid" since the list indentation is very weird. Would it make more sense to have more "normal" test cases?

```suggestion
      await testValidExample([
         '> text',
         '>',
         '>    * text',
         '>',
         '>    1. text',
      ]);
      await testValidExample([
         '> text',
         '>',
         '>    1. text',
         '>',
         '>    * text',
      ]);
      await testValidExample([
         '> text',
         '>',
         '>    * text',
         '>',
         '>> text',
         '>>',
         '>>    * text',
      ]);
      await testValidExample([
         '> text',
         '>',
         '>    1. text',
         '>',
         '>> text',
         '>>',
         '>>    1. text',
      ]);
      await testValidExample([
         '> text',
         '>',
         '>    * text',
         '>',
         '>> text',
         '>>',
         '>>    1. text',
      ]);
      await testValidExample([
         '> text',
         '>',
         '>    1. text',
         '>',
         '>> text',
         '>>',
         '>>    * text',
      ]);
```

File: src/indent-alignment.js [hunk 5]
```diff
@@ -172,6 +173,14 @@ module.exports = {
                return;
             }
 
+            // We don't need to enforce a specific list indentation inside blockQuote
+            // tokens. If we did, it would cause false negatives and conflict with
+            // existing rules. Additionally, we already enforce that the indentation for
```
Comment (onebytegone) [inline src/indent-alignment.js]: When making a statement like "it would cause false negatives and conflict with existing rules", it's usually a good idea to provide an example. Not doing so creates situations like: https://xkcd.com/979

File: src/indent-alignment.js [hunk 6]
```diff
@@ -172,6 +173,14 @@ module.exports = {
...
+            // consecutive blocks of tokens be of equal length, which is sufficient.
```
Comment (onebytegone) [inline src/indent-alignment.js]: Perhaps something like:
```suggestion
            // This rule does not enforce a specific initial indentation of list items
            // since that behavior is handled by rules like MD007. Therefore, ignore any
            // list tokens that are the direct children of blockQuote tokens.
```

This also makes me wonder, would it be more clear to simply not iterate over list tokens when iterating over the children of blockquotes rather than skipping them?

**Review by onebytegone — CHANGES_REQUESTED** [review]

File: src/indent-alignment.js (hunk 5 above)
Comment (MrMarCode) [reply src/indent-alignment.js]: That comic is hilarious! Thanks for the good suggestion.

File: src/indent-alignment.js [hunk 7]
```diff
@@ -118,13 +119,24 @@ module.exports = {
...
 
+         // This rule does not enforce a specific initial indentation of list items
+         // since that behavior is handled by rules like MD007. Therefore, ignore any
+         // list tokens that are the direct children of blockQuote tokens.
+         // We also ignore nested blockQuotes because the checks below are not written
+         // to handle that edge case.
+         if (token.type === 'blockQuote') {
+            iterateOverChildTokens.pop();
+            iterateOverChildTokens.pop();
+            iterateOverChildTokens.pop();
```
Comment (onebytegone) [inline src/indent-alignment.js]: Popping items off a list like this is extremely fragile. All it takes is someone to reorder `iterateOverChildTokens` and the code no longer works. A couple more reliable approaches are (1) `filter` the array to exclude the items you don't want or (2) use different lists depending on the `token.type`.

File: src/indent-alignment.js [hunk 8]
```diff
@@ -118,13 +119,24 @@ module.exports = {
...
 
          if (!firstToken) {
             return;
          }
 
+         // This rule does not enforce a specific initial indentation of list items
+         // since that behavior is handled by rules like MD007. Therefore, ignore any
+         // list tokens that are the direct children of blockQuote tokens.
+         // We also ignore nested blockQuotes because the checks below are not written
+         // to handle that edge case.
```
Comment (onebytegone) [inline src/indent-alignment.js:134]: We need at least one test to ensure we don't end up with this same issue again.

**Review by onebytegone — CHANGES_REQUESTED** [review]

**Review by onebytegone — APPROVED** [review]
